spec.md

```diff
@@ -0,0 +1,1813 @@
+* it's type assignment satisfies `P` as described in the previous section.
+* it's concept assignment satisfies `P` by satisfying _each statement in `P`_ ... as we now describe.
+
+#### Types
+
+**Case TYPE_DEF**
+* `Kind $A` (for `Kind` in `{entity, relation, attribute}`) is satisfied if $`m(A) : \mathbf{Kind}`$
+
+* `(Kind) $A sub $B` is satisfied if $`m(A) : \mathbf{Kind}`$, $`m(B) : \mathbf{Kind}`$, $`m(A) \lneq m(B)`$
+* `(Kind) $A sub! $B` is satisfied if $`m(A) : \mathbf{Kind}`$, $`m(B) : \mathbf{Kind}`$, $`m(A) <_! m(B)`$
+
+_Remark_: `sub!` is convenient, but could actually be expressed with `sub`, `not`, and `is`. Similar remarks apply to **all** other `!`-variations of TypeQL key words below.
+
+**Case REL_PATT**
+* `$A relates $I` is satisfied if $`m(A) : \mathbf{Rel}(m(I))`$
+
+* `$A relates! $I` is satisfied if $`m(A) : \mathbf{Rel}(m(I))`$ and **not** $`m(A) \lneq m(B) : \mathbf{Rel}(m(I))`$
+* `$A relates $I as $J` is satisfied if $`m(A) : \mathbf{Rel}(m(I))`$, $`B : \mathbf{Rel}(m(J))`$, $`A < B`$, $m(I) < m(J)$.
+* `$A relates $I[]` is satisfied if $`m(A) : \mathbf{Rel}(m([I]))`$
+* `$A relates! $I[]` is satisfied if $`m(A) : \mathbf{Rel}(m([I]))`$ and **not** $`m(A) \lneq m(B) : \mathbf{Rel}(m([I]))`$
+* `$A relates $I[] as $J[]` is satisfied if $`m(A) : \mathbf{Rel}(m([I]))`$, $`B : \mathbf{Rel}(m([J]))`$, $`A < B`$, $m(I) < m(J)$.
+
+**Case PLAY_PATT**
+* `$A plays $I` is satisfied if $`m(A) < A' <_! m(I)`$ (for $A'$ **not** an interface type)
+* `$A plays! $I` is satisfied if $`m(A) <_! m(I)`$
+
+**Case OWNS_PATT**
+* `$A owns $B` is satisfied if $`m(A) < A' <_! m(O_B)`$ (for $A'$ **not** an interface type)
+* `$A owns! $B` is satisfied if $`m(A) <_! m(O_B)`$ 
+
+_Remark_. In particular, if `A owns B[]` has been declared, then `$X owns B` will match the answer `m($X) = A`.
+
+#### Constraints
+
+_Remark: the usefulness of constraint patterns seems overall low, could think of a different way to retrieve full schema or at least annotations (this would be more useful than, say,having to find cardinalities by "trialing and erroring" through matching). STICKY: discuss!_
+
+**Case CARD_PATT**
+* cannot match `@card(n..m)` (STICKY: discuss! `@card($n..$m)`??)
+<!-- 
+* `A relates I @card(n..m)` is satisfied if $`m(A) : \mathbf{Rel}(m(I))`$ and schema allows $|a|_I$ to be any number in range `n..m`.
+* `A plays B:I @card(n..m)` is satisfied if ...
+* `A owns B @card(n...m)` is satisfied if ...
+* `$A relates $I[] @card(n..m)` is satisfied if ...
+* `$A owns $B[] @card(n...m)` is satisfied if ...
+-->
+
+**Case PLAYS_AS_PATT**
+
+_Notation: for readability, we simply write $X$ in place of $m(X)$ in this case and the next._
+
+* `$A plays $B:$I as $C:$J` is satisfied if $A \leq A' <_! D' \leq D$ for some $D$s, and $I \leq I' <_! J' \leq J$, with $`A^{(')} < {I^{(')}}`$, $`D^{(')} < {J^{(')}}`$, and schema directly contains the constraint `A' plays B':I' as C':J'` for relation types $B \leq B' \leq_! C' \leq C$.
```

Comment:
This line, and the following line re `owns as` are not rendered in a legible manner!